basic/js/input.js

```diff
@@ -0,0 +1,75 @@
+const emailInput = document.querySelector('#email');
+const passwordInput = document.querySelector('#password');
+const passwordCheckInput = document.querySelector('#password_check');
+const toggleBtns = document.querySelectorAll('.btn-visible');
+
+function showError(el, msg){
+    el.classList.add('input-error');
+    const parent = el.parentElement;
+    if (parent.querySelector('.txt-help')) {
+        return;
+    }
+    const errorEl = document.createElement('p')
+    errorEl.classList.add('txt-help', 'error');
+    errorEl.textContent = msg;
+    parent.appendChild(errorEl)
+}
+function clearError(el){
+    el.classList.remove('input-error');
+
+    const parent = el.parentElement;
+    const errorEl = parent.querySelector('.txt-help');
+    if (errorEl) {
+        parent.removeChild(errorEl);
+    }
+}
+
+function emailValidate(){
+    const emailRegex = /^[^\s@]+@[^\s@]+\.[^\s@]+$/;
+
+    if (!emailInput.value) {
+        showError(emailInput, '이메일을 입력해주세요.');
+    }else if (!emailRegex.test(emailInput.value)) {
+        showError(emailInput, '잘못된 이메일 형식입니다.');
+    } else{
+        clearError(emailInput);
+    }
```

Comment:
```suggestion
    if (!emailInput.value) {
        showError(emailInput, '이메일을 입력해주세요.');
        return;
    }
    if (!emailRegex.test(emailInput.value)) {
        showError(emailInput, '잘못된 이메일 형식입니다.');
        return
    }
    clearError(emailInput);
}
```
이런식으로 early return 패턴을 이용해도 좋습니다.